src/resdk/resources/data.py

```diff
@@ -156,10 +156,14 @@ def descriptor_schema(self, payload):
     def sample(self):
         """Get sample."""
         if self._sample is None and self._original_values.get("entity", None):
+            # The collection data is only serialized on the top level. Replace the
+            # data inside 'entity' with the actual collection data.
+            self._original_values["entity"]["collection"] = self._original_values.get(
```

Comment:
Do you know if we need this? If not, I would prefer to keep the "original data" actually original.

Reply:
OK, I will construct another dict and avoid modifying the original values.